packages/auth/src/core/auth/auth_impl.ts

```diff
@@ -223,6 +225,10 @@ export class AuthImpl implements AuthInternal, _FirebaseService {
     _assert(this._popupRedirectResolver, this, AuthErrorCode.ARGUMENT_ERROR);
     await this.getOrInitRedirectPersistenceManager();
 
+    // At this point in the flow, this is a redirect user. Run blocking
+    // middleware callbacks before setting the user.
+    await this._runBeforeStateCallbacks(storedUser);
```

Comment:
I think here we should wrap it in a try/catch. Initialization swallows errors. Ideally we would surface the error in `getRedirectResult` but that will require some extra thinking